src/test/scala/za/co/absa/hyperdrive/trigger/persistance/JobTemplateHistoryRepositoryTest.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.hyperdrive.trigger.persistance
+
+import org.scalatest.{FlatSpec, _}
+import za.co.absa.hyperdrive.trigger.models.History
+import za.co.absa.hyperdrive.trigger.models.enums.DBOperation
+import za.co.absa.hyperdrive.trigger.models.enums.DBOperation.DBOperation
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class JobTemplateHistoryRepositoryTest extends FlatSpec with Matchers with BeforeAndAfterAll with BeforeAndAfterEach with RepositoryH2TestBase {
+  import api._
+
+  private val h2JobTemplateHistoryRepository: JobTemplateHistoryRepository =
+    new JobTemplateHistoryRepositoryImpl(dbProvider) with H2Profile  {
+      override val profile = h2Profile
+    }
+
+  private val h2jobTemplateHistoryTable = h2JobTemplateHistoryRepository.jobTemplateHistoryTable
+
+  override def beforeAll: Unit = {
+    schemaSetup()
+  }
+
+  override def afterAll: Unit = {
+    schemaDrop()
+  }
+
+  override def afterEach: Unit = {
+    clearData()
+  }
+
+  private def verifyHistory(historyEntries: Seq[History], historyId: Long, user: String, dbOperation: DBOperation): Boolean = {
+    historyEntries.exists(history => {
+      history.id == historyId && history.changedBy == user && history.operation == dbOperation
+    })
+  }
+
+  "notificationRuleHistoryRepository create/update/delete/getHistoryForWorkflow/getJobTemplatesFromHistory" should "create workflow history record with correct values" in {
```

Comment:
`notificationRuleHistoryRepository` caught you copying :D

Reply:
Fixed